sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1250,7 +1252,38 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
    */
   override def visitTableName(ctx: TableNameContext): LogicalPlan = withOrigin(ctx) {
     val tableId = visitMultipartIdentifier(ctx.multipartIdentifier)
-    val table = mayApplyAliasPlan(ctx.tableAlias, UnresolvedRelation(tableId))
+    val properties = new util.HashMap[String, String]
+    if (ctx.asOf != null && ctx.asOf.version != null) {
+      properties.put(TableCatalog.PROP_VERSION, ctx.asOf.version.getText)
+    } else if (ctx.asOf != null && ctx.asOf.TIMESTAMP != null) {
+      val ts = ctx.asOf.timestamp.getText
+      if (ts.length == "'yyyy-MM-dd'".length) {
+        val df = DateFormatter()
+        try {
+          properties.put(
+            TableCatalog.PROP_TIMESTAMP,
+            df.parse(ts.substring(1, ts.length - 1)).toString)
+        } catch {
+          case _: Throwable =>
+            throw new IllegalArgumentException(s"Illegal timestamp value $ts in TIMESTAMP AS OF")
```

Comment:
Should we also include original `Throwable` inside the new `IllegalArgumentException`? It might help debug.

Reply:
added